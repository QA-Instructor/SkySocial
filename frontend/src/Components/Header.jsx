import React from 'react'
import { Link,Outlet } from "react-router-dom";
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import "bootstrap/dist/css/bootstrap.min.css"
import "../resources/header.css"


export const Header = () => {
  return (
        <>
        <header>
            <div className='header-logo-section'>
                <img alt='Sky Social'/>
            </div>

            <Navbar className='navbar'>
                <Nav className='nav'>
                    <ul className='nav-list-left'>
                        <li className='navbar-item'id='home'>
                            <Link className='nav-link' to='/'>Home</Link>
                        </li>
                        <li className='navbar-item' id='createEvent'>
                            <Link className='nav-link' to='/CreateEvent'>Create Event</Link>
                        </li>
                        <li className='navbar-item' id='myEvents'>
                            <Link className='nav-link' to='/MyEvents'>My Events</Link>
                        </li>
                        
                    </ul>
                    <ul className='nav-list-right'>
                    <li className='navbar-item' id='search'>
                            <Link className='nav-link' to='/Search'>Search</Link>
                        </li>
                        <li className='navbar-item' id='profile'>
                            <Link className='nav-link' to='/Profile'>Profile</Link>
                        </li>
                        <li className='navbar-item' id='login'>
                            <Link className='nav-link' to='/Login'>Logout</Link>
                        </li>
                    </ul>
                </Nav>
            </Navbar>
        </header>
        <Outlet/>
        </>
  )
}
export default Header;
