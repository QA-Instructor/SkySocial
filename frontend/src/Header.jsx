import React from 'react'
import { Link } from "react-router-dom";
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import "bootstrap/dist/css/bootstrap.min.css"
import "./header.css"
export const Header = () => {
  return (
    <header>
        <div className='header-logo-section'>
            <img alt='Sky Social'/>
        </div>

        <Navbar className='navbar'>
            <Nav className='nav'>
                <ul className='nav-list-left'>
                    <li className='navbar-item'>
                        <Link className='nav-link' to='/home'>Home</Link>
                    </li>
                    <li className='navbar-item'>
                        <Link className='nav-link' to='/createEvent'>Create Event</Link>
                    </li>
                    <li className='navbar-item'>
                        <Link className='nav-link' to='/myEvents'>My Events</Link>
                    </li>
                    
                </ul>
                <ul className='nav-list-right'>
                <li className='navbar-item'>
                        <Link className='nav-link' to='/search'>Search</Link>
                    </li>
                    <li className='navbar-item'>
                        <Link className='nav-link' to='/profile'>Profile</Link>
                    </li>
                    <li className='navbar-item'>
                        <Link className='nav-link' to='/login'>Logout</Link>
                    </li>
                </ul>
            </Nav>
        </Navbar>
    </header>
  )
}
export default Header;
