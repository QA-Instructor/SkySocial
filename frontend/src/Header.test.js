import React from 'react';
import { create } from 'react-test-renderer';
import { screen, act, render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom';


import Header from './Header';

describe('Header Snapshot tests',()=>{
    test(`Header matches snapshot`, () => {
        const header = create(
            <BrowserRouter>
                <Header />
            </BrowserRouter>);
        expect(header.toJSON()).toMatchSnapshot();
    });
    test.skip(`Searchbar appears`, () => {
        const searchBar = create(<SearchBar />);
        expect(searchBar.toJSON()).toMatchSnapshot();
    });    
});

describe("Content appears as expected",()=>{
    test(`logo appears`, async () => {
        await act(async () => render(<BrowserRouter>
            <Header />
        </BrowserRouter>));
        expect(await screen.findByAltText('Sky Social')).toBeInTheDocument()
    });
    test(`home button appears`, async () => {
        await act(async () => render(<BrowserRouter>
            <Header />
        </BrowserRouter>));
        expect(await screen.findByText(/home/i)).toBeInTheDocument()
    });
    test(`Create Event button appears`, async () => {
        await act(async () => render(<BrowserRouter>
            <Header />
        </BrowserRouter>));
        expect(await screen.findByText(/Create Event/i)).toBeInTheDocument()
    });
    test(`My Events button appears`, async () => {
        await act(async () => render(<BrowserRouter>
            <Header />
        </BrowserRouter>));
        expect(await screen.findByText(/My Events/i)).toBeInTheDocument()
    });
    test(`Search button appears`, async () => {
        await act(async () => render(<BrowserRouter>
            <Header />
        </BrowserRouter>));
        expect(await screen.findByText(/Search/i)).toBeInTheDocument()
    });
})