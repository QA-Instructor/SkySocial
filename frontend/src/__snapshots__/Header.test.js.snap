// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot tests Header matches snapshot 1`] = `
<header>
  <div
    className="header-logo-section"
  >
    <img
      alt="Sky Social"
    />
  </div>
  <nav
    className="navbar navbar navbar-expand navbar-light"
  >
    <div
      className="nav navbar-nav"
      onKeyDown={[Function]}
    >
      <ul
        className="nav-list-left"
      >
        <li
          className="navbar-item"
          id="home"
        >
          <a
            className="nav-link"
            href="/home"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="navbar-item"
          id="createEvent"
        >
          <a
            className="nav-link"
            href="/createEvent"
            onClick={[Function]}
          >
            Create Event
          </a>
        </li>
        <li
          className="navbar-item"
          id="myEvents"
        >
          <a
            className="nav-link"
            href="/myEvents"
            onClick={[Function]}
          >
            My Events
          </a>
        </li>
      </ul>
      <ul
        className="nav-list-right"
      >
        <li
          className="navbar-item"
          id="search"
        >
          <a
            className="nav-link"
            href="/search"
            onClick={[Function]}
          >
            Search
          </a>
        </li>
        <li
          className="navbar-item"
          id="profile"
        >
          <a
            className="nav-link"
            href="/profile"
            onClick={[Function]}
          >
            Profile
          </a>
        </li>
        <li
          className="navbar-item"
          id="login"
        >
          <a
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Logout
          </a>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
